sklearn/compose/_column_transformer.py

```diff
@@ -459,7 +459,10 @@ def _check_key_type(key, superclass):
     if isinstance(key, list):
         return all(isinstance(x, superclass) for x in key)
     if hasattr(key, 'dtype'):
-        return key.dtype.kind == 'i'
+        if superclass is int:
```

Comment:
Sorry I haven't reviewed this part of the code but this function seems pretty opaque. Can you please add a proper docstring? What is superclass and why is it ignored if ``key`` has ``dtype``?

Reply:
Yes, will add a proper docstring. Superclass is actually only either `int` or `six.string_types`. Therefore the if/else in case it has a dtype

Reply:
yeah I saw that afterwards, but in isolation it's a bit hard to understand.